service/src/test/java/org/cbioportal/genome_nexus/component/annotation/CanonicalTranscriptResolverTest.java

```diff
@@ -130,4 +133,39 @@ public void resolveCanonicalTranscript() throws IOException
             this.canonicalTranscriptResolver.resolve(variantMockData.get("4:g.55593656_55593657insCAACTTCCTTATGATCACAAATGGGAGTTTCCCAGAAACAGGCTGAGTTTTGGT"))
         );
     }
+
+    @Test
+    public void resolveCanonicalTranscriptWithOverrides() throws IOException
+    {
+        Map<String, VariantAnnotation> variantMockData = this.variantAnnotationMockData.generateData();
+
+        // empty override list, should have the same effect with no list
+        assertEquals(
+            "ENST00000424053",
+            this.canonicalTranscriptResolver
+                .resolve(variantMockData.get("3:g.14106026_14106037del"), Collections.emptySet())
+                .getTranscriptId()
+        );
+
+        // override canonical transcripts with just one matching transcript
+        String[] overridesWithSingleMatch = {"ENST00000429201"};
+
+        assertEquals(
+            "ENST00000429201",
+            this.canonicalTranscriptResolver
+                .resolve(variantMockData.get("3:g.14106026_14106037del"), new HashSet<>(Arrays.asList(overridesWithSingleMatch)))
+                .getTranscriptId()
+        );
+
+        // override canonical transcripts with more than one matching transcript
+        String[] overridesWithMultipleMatch = {"ENST00000532753", "ENST00000532924", "ENST00000532880"};
+
+        // should pick the most impactful one (inframe_deletion)
+        assertEquals(
+            "ENST00000532924",
+            this.canonicalTranscriptResolver
+                .resolve(variantMockData.get("3:g.14106026_14106037del"), new HashSet<>(Arrays.asList(overridesWithMultipleMatch)))
+                .getTranscriptId()
+        );
+    }
```

Comment:
Maybe good to add this example as well `9:g.21994291C>T`. Could be an integration test

Reply:
We can add the actual data for `9:g.21994291C>T`, but the override lists here are random though.